sonaranalyzer-dotnet/tests/SonarAnalyzer.UnitTest/TestCases/IndentSingleLineFollowingConditional.cs

```diff
@@ -0,0 +1,310 @@
+//         ^^^^^^^^^^^^^^^^^^ Secondary
+
+
+            for(int i = 0; i < 10; i++) // Noncompliant {{Use curly braces or indentation to denote the code conditionally executed by this 'for'}}
+//          ^^^^^^^^^^^^^^^^^^^^^^^^^^^
+       total = total + i;               // trivia not included in secondary location for single line statements...
+//     ^^^^^^^^^^^^^^^^^^ Secondary
+
+
+            foreach(char item in data)  // Noncompliant {{Use curly braces or indentation to denote the code conditionally executed by this 'foreach'}}
+//          ^^^^^^^^^^^^^^^^^^^^^^^^^^
+            total++;                    // trivia not included in secondary location for single line statements...
+//          ^^^^^^^^ Secondary
+
+
+            if (total < 100)            // Noncompliant {{Use curly braces or indentation to denote the code conditionally executed by this 'if'}}
+//          ^^^^^^^^^^^^^^^^
+            total = 100;                // trivia not included in secondary location for single line statements...
+//          ^^^^^^^^^^^^ Secondary
+            else                        // Noncompliant {{Use curly braces or indentation to denote the code conditionally executed by this 'else'}}
+//          ^^^^
+            total = 200;
+//          ^^^^^^^^^^^^ Secondary
+        }
+
+
+        public int SpecialElseIfCase(int i)
+        {
+            if (i > 100)
+                return 1;
+            else if (i > 200)
+                return 2;  // compliant - common pattern used in Akka, and used Ember and Nancy too.
+
+
+            if (i > 300)
+                return 3;
+            else if (i > 400) // Noncompliant {{Use curly braces or indentation to denote the code conditionally executed by this 'else if'}}
+//          ^^^^^^^^^^^^^^^^^
+            return 4;
+//          ^^^^^^^^^ Secondary
+
+            if (i > 400)
+                return 4;
+            else if (i > 500) // Noncompliant {{Use curly braces or indentation to denote the code conditionally executed by this 'else if'}}
+//          ^^^^^^^^^^^^^^^^^
+      return 5;
+//    ^^^^^^^^^ Secondary
+
+            return 6;
+        }
+
+        public int OtherElseIfCombinations(int i)
+        {
+            if (i > 100)
+                return 1;
+            else
+//          ^^^^
+            if (i > 200)
+//          ^^^^^^^^^^^^ Secondary
+//          ^^^^^^^^^^^^ @-1
+            return 2;
+//          ^^^^^^^^^ Secondary
+
+            return 0;
+        }
+
+        public int Non_compliant_NestedMixedConditionals()
+        {
+            int total = 0;
+            string data = "abc";
+
+            do                
+//          ^^
+            while (total < 10)                  // comments are included in highlighting for compound statements
+//          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ Secondary
+//          ^^^^^^^^^^^^^^^^^^ @-1
```

Comment:
It is not obvious if this is a primary location, e.g. usually marked with a `// Noncompliant` comment. As far as I can remember you could do `// ^^^^^^^^ Noncompliant @-1`

Reply:
Thanks, I'll try it. Originally I had _Noncompliant_ on the line containing the error, but the test framework complained that a precise location was being specified more than once and the _Noncompliant_ should be removed. I didn't think about trying it on the ^^^ line.

Reply:
Update: `// ^^^^^ Noncompliant @-1`  didn't work. The error framework didn't complain, but the tests failed because it "the error wasn't expected" at the marked line.
It works correctly with ` // ^^^^ @-1`.  The issues is definitely being checked for correctly with that syntax (changing the number of ^ is enough to cause the test to fail because the location length isn't correct).